app/src/org/commcare/tasks/ResourceRecoveryTask.java

```diff
@@ -38,8 +53,55 @@ protected Boolean doTaskBackground(Void... voids) {
         return success;
     }
 
+    @Override
+    protected void onPostExecute(Boolean aBoolean) {
+        super.onPostExecute(aBoolean);
+        clearTaskInstance();
```

Comment:
Abstractly, I think this (and the call below) should be in a finally() {}, but all of the locking code right now is in areas that'd only execute on the UI thread, so it's hard for me to come up with a clear narrative where the onX() callbacks can crash and the code continues executing.

It may make more sense (based on the reasoning I mentioned above) for the clearing to actually happen in a finaly() at the end of execution, rather than in the callback. That would prevent the state of the task from changing in-between connection. You'd have to put the Activity attach() inside of the getNewIsntance() for that to work.

Reply:
Just for context I was trying to use the same pattern `SingletonTask` [uses](https://github.com/dimagi/commcare-android/blob/master/app/src/org/commcare/tasks/SingletonTask.java#L38) . When you say we should move clearing at `end of execution` do you mean something like this - 

```
    @Override
    protected Boolean doTaskBackground(Void... voids) {
        try {
         // do work 
        } finally {
        clearTaskInstance();
       }
        return success;
    }
```

Reply:
Ah, didn't realize that pattern was involved elsewhere. Let's stay consistent and manage it as outlined.